src/productbasis.jl

```diff
@@ -0,0 +1,61 @@
+# Jerry: This is just a specific case of a general ProductBasis
```

Comment:
At the moment, it feels like this type should be removed entirely and replaced with a convenience function that generates a Lux network. That network can contain the transforms that you need to solve the problem you mention here in the comments.

Reply:
> This is the first part of my review.
> 
> Thank you for you really excellent work on this!! I'll continue by looking at the unit tests now.

Thank you for your review. I will look into the issue later today.

Reply:
> At the moment, it feels like this type should be removed entirely and replaced with a convenience function that generates a Lux network. That network can contain the transforms that you need to solve the problem you mention here in the comments.

Yes. I just realize recently that there is a `Lux.Parallel` layer that allow us to do this easily. I will also remove that and replace that with a function that returns a `Lux.Chain`. It should be fine to be removed since we are not using this elsewhere other than `ACEpsi.jl`.

Reply:
A Problem here is that if we want this functionality we have to include Lux in our dependency since we have to make use of `Lux.Parallel` and `Lux.Chain`. I am not sure whether we should have that.

@cortner Please advice.

Reply:
Good point. I think i agree with you. Remove Lux but add an example